BackendIntegrationTests/BackendIntegrationTests.swift

```diff
@@ -215,23 +215,62 @@ class BackendIntegrationSK1Tests: XCTestCase {
 
         expect(customerInfo.entitlements.all.count) == 1
         let entitlements = self.purchasesDelegate.customerInfo?.entitlements
-        expect(entitlements?["premium"]?.isActive) == true
+        expect(entitlements?[Self.entitlementIdentifier]?.isActive) == true
 
         let anonUserID = Purchases.shared.appUserID
         let identifiedUserID = "\(#function)_\(anonUserID)_".replacingOccurrences(of: "RCAnonymous", with: "")
 
         let (identifiedCustomerInfo, created) = try await Purchases.shared.logIn(identifiedUserID)
         expect(created) == true
-        expect(identifiedCustomerInfo.entitlements["premium"]?.isActive) == true
+        expect(identifiedCustomerInfo.entitlements[Self.entitlementIdentifier]?.isActive) == true
 
         eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility([productID])
         expect(eligibility[productID]?.status) == .ineligible
     }
 
+    func testUserHasNoEligibleOffersByDefault() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        expect(product.discounts).to(haveCount(1))
+        expect(product.discounts.first?.offerIdentifier) == "com.revenuecat.monthly_4.99.1_free_week"
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(beEmpty())
+    }
+
+    func testPurchaseWithDiscount() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        try await self.purchaseMonthlyOffering()
+        try self.verifyEntitlementWentThrough()
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(haveCount(1))
+        let offer = try XCTUnwrap(offers.first)
+
+        _ = try await Purchases.shared.syncPurchases()
+
+        _ = try await Purchases.shared.purchase(product: product, promotionalOffer: offer)
+
+        let entitlement = try self.verifyEntitlementWentThrough()
+        expect(entitlement.willRenew) == true
+        expect(entitlement.periodType) == .intro
```

Comment:
🤔 I think you can still have intro pricing without a promo, right? 
Like, you can have a regular purchase be $1.99 for 3 months, $6.99 after, and without any promo offers, and this would be true

Reply:
does the purchase date get updated? Or otherwise, perhaps there's another proxy we can use for this? 

Also, I think we should add a comment or something explicitly explaining that we're using this as a proxy for figuring out whether a promotional offer was applied

Reply:
does the purchase date get updated? Or otherwise, perhaps there's another proxy we can use for this? 

Also, I think we should add a comment or something explicitly explaining that we're using this as a proxy for figuring out whether a promotional offer was applied

Reply:
Oh that's right, I misinterpreted this. 
Looks like the latest purchase date doesn't get updated :/ so I still can't see what we can use to check this.

I'm taking another look...

Reply:
Yup, still not seeing any changes. I've added a new product that doesn't have a 1 week free intro, and I still see the same.
Before the purchase with the promo:
```
<EntitlementInfo: "
identifier=premium,
isActive=true,
willRenew=true,
periodType=PeriodType(rawValue: 0),
latestPurchaseDate=Optional(2022-03-08 21:42:09 +0000),
originalPurchaseDate=Optional(2022-03-08 21:42:09 +0000),
expirationDate=Optional(2022-03-08 21:42:40 +0000),
store=Store(rawValue: 0),
productIdentifier=com.revenuecat.monthly_4.99.no_intro,
isSandbox=true,
unsubscribeDetectedAt=nil,
billingIssueDetectedAt=nil,
ownershipType=PurchaseOwnershipType(rawValue: 0)
>
```
After:
```
<EntitlementInfo: "
identifier=premium,
isActive=true,
willRenew=true,
periodType=PeriodType(rawValue: 0),
latestPurchaseDate=Optional(2022-03-08 21:42:09 +0000),
originalPurchaseDate=Optional(2022-03-08 21:42:09 +0000),
expirationDate=Optional(2022-03-08 21:42:40 +0000),
store=Store(rawValue: 0),
productIdentifier=com.revenuecat.monthly_4.99.no_intro,
isSandbox=true,
unsubscribeDetectedAt=nil,
billingIssueDetectedAt=nil,
ownershipType=PurchaseOwnershipType(rawValue: 0)
>
```